### PR TITLE
[ISSUE-#4256] Just inject environment in StartingSpringApplicationRunListener

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change

For-#4256

### CODE REVIEW

**Comment by chuntaojun** [comment]
SpringApplicationListener If there is a concept of priority, we need to see

**Comment by chuntaojun** [comment]
please resolve code conflict

**Comment by chuntaojun** [comment]
this is no good solution,

class StartingSpringApplicationListener {
    List<NacosApplicationListener> listeners;


    func start() {
         for each listener => listener.onStart()
    }
}

**Comment by horizonzy** [comment]
> ```java
> listeners
> ```

good idea.

File: core/src/main/java/com/alibaba/nacos/core/listener/NacosApplicationListener.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+package com.alibaba.nacos.core.listener;
+
+import com.alibaba.nacos.core.code.SpringApplicationRunListener;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Nacos Application Listener, execute init process.
+ */
+public interface NacosApplicationListener {
```
Comment (chuntaojun) [inline core/src/main/java/com/alibaba/nacos/core/listener/NacosApplicationListener.java:26]: 将来可以考虑这里面不要依赖Spring的对象信息

**Review by chuntaojun — APPROVED** [review]

File: core/src/main/java/com/alibaba/nacos/core/listener/NacosApplicationListener.java (hunk 1 above)
Comment (horizonzy) [reply core/src/main/java/com/alibaba/nacos/core/listener/NacosApplicationListener.java:26]: 恩。现在还是依赖的spring, 需要用里面的environment, 以后如果要剔除spring时，再进行重构。

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**